### PR TITLE
ROCM: fixed build on AMD systems

### PR DESCRIPTION
**Pull request by hoopoepg**
- there was issue in ROCM implementation due to errors in implementation
  of new MD API
- fixed

fixes https://github.com/openucx/ucx/issues/6865

### CODE REVIEW

File: src/uct/rocm/copy/rocm_copy_md.c [hunk 1]
```diff
@@ -268,9 +266,13 @@ static void uct_rocm_copy_rcache_mem_dereg_cb(void *context, ucs_rcache_t *rcach
 {
     uct_rocm_copy_md_t *md = context;
     uct_rocm_copy_rcache_region_t *region;
+    uct_md_mem_dereg_params_t params;
 
-    region = ucs_derived_of(rregion, uct_rocm_copy_rcache_region_t);
-    (void)uct_rocm_copy_mem_dereg(&md->super, &region->memh);
+    region            = ucs_derived_of(rregion, uct_rocm_copy_rcache_region_t);
```
Comment (yosefe) [inline src/uct/rocm/copy/rocm_copy_md.c:271]: init inline

**Review by souravzzz — APPROVED** [review]

**Comment by hoopoepg** [comment]
infra issues

File: src/uct/rocm/copy/rocm_copy_md.c (hunk 1 above)
Comment (hoopoepg) [reply src/uct/rocm/copy/rocm_copy_md.c]: fixed

**Comment by hoopoepg** [comment]
unrelated issue

**Comment by hoopoepg** [comment]
bot:pipe:retest

File: src/uct/rocm/copy/rocm_copy_md.c [hunk 2]
```diff
@@ -267,10 +265,14 @@ static void uct_rocm_copy_rcache_mem_dereg_cb(void *context, ucs_rcache_t *rcach
                                               ucs_rcache_region_t *rregion)
 {
     uct_rocm_copy_md_t *md = context;
-    uct_rocm_copy_rcache_region_t *region;
+    uct_rocm_copy_rcache_region_t *region =
```
Comment (yosefe) [inline src/uct/rocm/copy/rocm_copy_md.c:268]: minor: pls align on column

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
replaced by #6851

**Pull request closed without merging**